staging/vhost-device-sound/src/stream.rs

```diff
@@ -254,31 +252,37 @@ impl std::fmt::Debug for Buffer {
 impl Buffer {
     pub fn new(data_descriptor: virtio_queue::Descriptor, message: Arc<IOMessage>) -> Self {
         Self {
+            bytes: vec![],
             pos: 0,
             data_descriptor,
+            populated: false,
             message,
         }
     }
 
-    pub fn consume(&self, buf: &mut [u8]) -> Result<u32> {
-        let addr = self.data_descriptor.addr();
-        let offset = self.pos as u64;
-        let len = self
+    pub fn prepare_output(&mut self) -> Result<()> {
+        if self.populated {
+            return Ok(());
+        }
+        self.bytes = vec![0; self.data_descriptor.len() as usize];
```

Comment:
Why does `prepare_output()` allocate an intermediate new buffer instead of reading from guest memory to the destination buffer as it is done in the previous version of the function?

Reply:
Thanks, I hadn't noticed that. Will push a new version tomorrow with everything mentioned so far.

Reply:
This has been resolved.